### PR TITLE
161 bug testing trpc got hands code coverage for api folder

### PR DESCRIPTION
**Pull request by thienguen**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [x] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [x] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [x] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [x] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?

- [ ] Fixed trpc context, allowing us to have server-side caller
- [ ] ALLOW TO RUN API FOR CODE COVERAGE
- [ ] Rename github action
- [ ] Dependencies control for test, and jest. Get it out of root and implement it into the designated API and expo folder, we will be testing these 2 as they are our product code 
- [ ] Beside those 2 folders, the rest are template and config code which no body tests on.
- [ ] Reference directly from the docs https://trpc.io/docs/server/server-side-calls
- [ ] We still keep trpc 11
- [ ] Also include a UI when testing
- [ ] We can leverage both vitest and jest under our repo, pretty cool.

# Tests

How was this tested?

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

Explanation:

- [ ] Follow these steps and make sure you received the same output as the screenshots below
- [ ] pnpm i (at root)
- [ ] pnpm db:prisma
- [ ] cd apps/expo
- [ ] pnpm run test 
- [ ] That will trigger jest 

image

- [ ] You should received this 
- [ ] now go back to root
- [ ] Make sure that your db instance is not production, use 2, or 3 or 4 (give a ping when ready) in .env at root
- [ ] pnpm db:seed

image

- [ ] cd packages/api
- [ ] We have 2 options to fk around here
- [ ] jest or vitest
- [ ] pnpm run jest 
- [ ] You should pass all the old tests (written for prisma) and a folder that gitignore away called `jest-coverage`

image

- [ ] comes back at root and seed db once again
- [ ] cd packages/api
- [ ] pnpm run vitest
- [ ] You should pass and 2 new folders will be generated, html and vitest-coverage 

image

- [ ] still at packages/api run
- [ ] pnpm run vitest-ui
- [ ] new local host exposed should spotted out

image

- [ ] Navigate there
- [ ] This is our test portal, if any failed, we will have better visualization of wtf happened

image

- [ ] End.

### CODE REVIEW

File: .github/workflows/cov.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   # > Jobs 1
-  build-n-test:
+  buil-n-coverage:
```
Comment (mva919) [inline .github/workflows/cov.yml:11]: Typo

**Review by mva919 — APPROVED** [review]
Fix the typo and then you can merge. Thanks for fixing trpc testing 🙏

File: .github/workflows/cov.yml (hunk 1 above)
Comment (thienguen) [reply .github/workflows/cov.yml:11]: kkk

**Review by thienguen — COMMENTED** [review]
merged reqady

**Pull request merged**